### PR TITLE
Display version information in the editor at all times

### LINKED ISSUE #27811: Display version number on the interface at all times by default
As a beginner, I'm having trouble following tutorials because subtle differences between versions create errors and uncertainty that could be easily avoided if I were able to reproduce their exact environment. This is a problem for noobs because our debugging skills are not very high yet. I [posted on Reddit](https://old.reddit.com/r/godot/comments/banp5k/suggestion_to_tutorial_makers_always_state_the/) about this and the thread got some traction. Ideally, tutorial makers should inform the version of Godot they are using, but if the version number was displayed by default the problem would be instantaneously solved.

### PR DESCRIPTION
**Pull request by Calinou**
This closes #27811.

## Preview

image

### CODE REVIEW

**Comment by akien-mga** [comment]
> PS: This causes the 2D/3D/Script/AssetLib buttons to be shifted towards the left when the version identifier is displayed. How could this be solved? I'd happily increase the display threshold to ~1250 pixels, in exchange of not having those buttons move when the version identifier becomes visible/hidden.

Wouldn't it be a better fit at the right side of the Bottom panel? There's more space available there and I think it's better to have version info in this lower corner than prominent next to the most important buttons.

**Comment by aaronfranke** [comment]
Also, I don't think it needs to be nearly as long as this. It can be shortened to "3.2-dev" etc. The full version string with hash is fine for debugging in the console, but not if it's always shown.

The big use case would be for tutorials where you can see the version in the video.

Also, another option is to place this in the window title bar, so that it says "Godot 3.2" etc.

**Comment by Calinou** [comment]
> Also, another option is to place this in the window title bar, so that it says "Godot 3.2" etc.

The window title bar isn't always visible in tutorials (e.g. if the video was recorded without window borders).

**Comment by YeldhamDev** [comment]
Maybe the full hash could be displayed as a tooltip when hovering it.

**Comment by Geequlim** [comment]
Why not put this in the about dialog ?

**Comment by akien-mga** [comment]
> Why not put this in the about dialog ?

It's already in the about dialog. I think the idea here is to have it always visible on screen (especially on screencasts).

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -44,6 +44,7 @@
 #include "editor_scale.h"
 #include "editor_settings.h"
 #include "editor_themes.h"
+#include "main/main.h"
```
Comment (akien-mga) [inline editor/project_manager.cpp]: This should be moved out of `main`, we shouldn't include `main.h` unless strictly necessary.

**Comment by Xrayez** [comment]
> I think the idea here is to have it always visible on screen (especially on screencasts).

I'm always compiling from source and having version information available on screen is really valuable. I'm also using game title as a custom build name which is nice to see there. Sometimes I have two Godot editors opened using different versions, so need a quick way to differentiate between them.

Also I'm thinking maybe it would be good to be able to copy the version string somehow just by clicking on it, this would allow users to include exact commit hash in bug reports quickly.

**Comment by Calinou** [comment]
I removed the changes related to code duplication, as these can be done in a later PR.

**Comment by Calinou** [comment]
I put the version label in the bottom panel per @akien-mga's suggestion:

image

It's also simpler to implement this way, and it displays at any window width.

**Comment by akien-mga** [comment]
> When a panel is active, the "Expand" button displays at the right. Is this the intended behavior, or should it be displayed at the left instead?

I think it's fine as is, I'd expect that button to stay at the rightmost position.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Sslaxx** [comment]
+1 for having the full version hash show up as a tooltip.

**Comment by Calinou** [comment]
@Sslaxx I guess this could be done now that the version identifier has been moved to the bottom panel.

**Comment by aaronfranke** [comment]
Should this be cherrypicked for 3.1? I think it would be useful to show the version at all times in currently supported versions, for people making tutorials in 2019.

**Comment by Calinou** [comment]
As mentioned by @aaronfranke, I think it'd be worth cherry-picking this to the `3.1` branch too.

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**